src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -211,6 +211,13 @@ private Task InternalWriteAsync(byte[] buffer, int offset, int count, Cancellati
 
         private Task<bool> InternalWriteDataAsync(byte[] buffer, int offset, int count, CancellationToken token)
         {
+            Debug.Assert(count >= 0);
+            
+            if (count == 0)
+            {
+                return Task.FromResult<bool>(true);
```

Comment:
Presumably this isn't a very common case? If it was, you could cache a true task and return that each time.